src/language-html/ast.js

```diff
@@ -84,6 +84,68 @@ class Node {
     fn(this);
   }
 
+  /**
+   * @param {Node} [target]
+   * @param {Object} [node]
+   */
+  insertChildBefore(target, node) {
+    const newNode = new Node(node);
+    setNonEnumerableProperties(newNode, {
+      // @ts-expect-error
+      index: target.index,
+      // @ts-expect-error
+      prev: target.prev,
+      next: target,
+      parent: this,
+    });
+
+    let next = target;
+    while (next) {
+      // @ts-expect-error
+      next.index++;
+      // @ts-expect-error
+      next = next.next;
+    }
```

Comment:
Is  it possible make this run into an infinite loop? Maybe we can recalculate `index`/`next` by looping children?